src/responder/pam/pamsrv.h

```diff
@@ -172,7 +175,24 @@ errno_t filter_responses(struct pam_ctx *pctx,
 
 errno_t pam_get_auth_types(struct pam_data *pd,
                            struct pam_resp_auth_type *_auth_types);
-errno_t pam_eval_prompting_config(struct pam_ctx *pctx, struct pam_data *pd);
+errno_t pam_eval_prompting_config(struct pam_ctx *pctx, struct pam_data *pd,
+                                  struct prompt_config ***_pc_list);
+
+/**
+ * @brief Get the prompt string for the selected type from the prompt_config
+ *        structure.
+ *
+ * @param[in] pc_list List that contains all authentication mechanisms prompts
+ * @param[in] prompt_type Authentication mechanism type
+ * @param[in] index String index, used for authentication mechanisms with more
+ *                  than one prompt
+ * @param[out] _prompt Prompt message
+ *
+ * @return 0 if the data was obtained properly,
+ *         error code otherwise.
+ */
+errno_t pam_get_prompting_option(struct prompt_config **pc_list, int prompt_type,
+                                 int index, const char **_prompt);
```

Comment:
Hi,

there already are
```
const char *pc_get_password_prompt(struct prompt_config *pc);
const char *pc_get_2fa_1st_prompt(struct prompt_config *pc);
const char *pc_get_2fa_2nd_prompt(struct prompt_config *pc);
const char *pc_get_2fa_single_prompt(struct prompt_config *pc);
const char *pc_get_passkey_inter_prompt(struct prompt_config *pc);
const char *pc_get_passkey_touch_prompt(struct prompt_config *pc);
```
in the client code. Imo it would be better to add `pc_get_eidp_init_prompt()` and `pc_get_eidp_link_prompt()` and reuse those functions in the PAM responder?

bye,
Sumit

Reply:
Makes sense, so I updated the implementation to take this into account.

Reply:
Thanks.